splight_lib/models/dashboard.py

```diff
@@ -59,67 +63,247 @@ class AdvancedFilter(SplightDatabaseBaseModel):
+            query_plain.extend(
+                [
+                    {
+                        "$addFields": {
+                            "timestamp": {
+                                "$dateTrunc": {
+                                    "date": "$timestamp",
+                                    "unit": self.query_group_unit,
+                                    "binSize": 1,
+                                }
+                            }
+                        }
+                    },
+                    {
+                        "$group": {
+                            "_id": "$timestamp",
+                            "value": {
+                                f"${self.query_group_function}": "$value"
+                            },
+                            "timestamp": {"$last": "$timestamp"},
+                        }
+                    },
+                ]
+            )
+        return json.dumps(query_plain)
+
+
+class Dashboard(SplightDatabaseBaseModel):
+    id: str | None = None
+    name: str
+    description: str | None = None
+    related_assets: List[ResourceSummary] | None = None
+    tags: List[ResourceSummary]
+
+
+class Tab(SplightDatabaseBaseModel):
+    id: str | None = None
+    name: str
+    order: int | None = None
+    dashboard: str
 
 
 class Chart(SplightDatabaseBaseModel):
     id: str | None = None
     name: str
-    description: Annotated[
-        str | None, Field(max_length=DESCRIPTION_MAX_LENGTH)
-    ] = None
-    items: list[ChartItem] | None = None
     tab: str
+    description: str | None = None
     position_x: int | None = None
     position_y: int | None = None
-    height: int | None = 28
-    width: int | None = 80
-    min_height: int | None = 14
-    min_width: int | None = 14
-    type: str | None = None
-    timestamp_gte: str | None = None
-    timestamp_lte: str | None = None
+    min_height: int | None = None
+    min_width: int | None = None
+    display_time_range: bool
+    labels_display: bool
+    labels_aggregation: str | None = None
+    labels_placement: str
     refresh_interval: str | None = None
     relative_window_time: str | None = None
-    y_axis_max_limit: int | None = None
-    y_axis_min_limit: int | None = None
-    config: dict[str, Any] | None = None
-    chart_items: list[ChartItem] = []
-    thresholds: list[Threshold] = []
+    show_beyond_data: bool
+    timezone: str | None = None
+    timestamp_gte: str | None = None
+    timestamp_lte: str | None = None
+    height: int
+    width: int
+    collection: str
+    chart_items: list[ChartItem]
+    thresholds: list[Threshold]
     value_mappings: list[
         ExactMatchValueMapping | RangeValueMapping | RegexMatchValueMapping
     ] = []
 
 
-class Tab(SplightDatabaseBaseModel):
-    id: str | None = None
-    name: str
-    charts: list[Chart] | None = None
-    order: int | None = None
-    dashboard: str
+class DashboardActionListChart(Chart):
+    type: str = "actionlist"
```

Comment:
Can this field be modified? If the value should be constant I would use Literal:
```python
class DashboardActionListChart(Chart):
    type: Literal["actionlist"] = "actionlist"
```

Reply:
Same for the rest of the charts